lamstare/experiments/run_test.py

```diff
@@ -1,91 +1,89 @@
+            job_name = f"IND_{run_name}"
+            command = (
+                f"#!/bin/bash \n"
+                f". /mnt/data_nas/public/.bashrc \n"
+                f"conda activate /mnt/data_nas/public/Miniconda/envs/{os.environ.get('CONDA_ENV','lamstare')} \n"
+                f"export PYTHONPATH=/mnt/data_nas/cc/LAMstare_new \n"
+                f"cd {Path(__file__).resolve().parent} \n"
+                f"python3 run_ind_test.py {exp_path} {head} {step} {testfile} {run_name} \n"
+            )
+            if query_job_numbers(job_name):
+                logging.warning(f"SKIPPED: {job_name} is already running.")
+            else:
+                logging.debug(f"Job command: \n{command}")
+                submit_job_to_dlc(job_name, command)
+                logging.info(f"SUBMITTED: {run_name}")
+        elif record_count == 1:
+            logging.info(f"SKIPPED: {run_name} already exists in DB.")
         else:
-            print(f"{temp_file_name} has multiple records, please check!!!")
+            logging.error(f"ERROR: {run_name} has multiple records, please check.")
 
 
-def find_ckpt_to_test_cron(exp_path:str, freq:int):
+def find_ckpt_to_test_cron(
+    exp_path: str, freq: int, record_type: type[Record]
+) -> Optional[int]:
     ckpt_pth = Path(exp_path)
-    with open(ckpt_pth / "checkpoint","r") as f:
-        latest_ckpt = int(f.readlines()[0].split("-")[1].split(".")[0])
-    # Get exp_path abs path
-    exp_path = str(ckpt_pth.resolve())
-    # Get basename
-    run_id = exp_path.split("/")[-1]
-    if len(Record.query(run_id=run_id)) > 0:
-        previous_tested_step = int(Record.query(run_id=run_id)[-1].step)
+    # model.ckpt-80000.pt -> 80000
+    with open(ckpt_pth / "checkpoint", "r") as f:
+        latest_ckpt_step = int(f.readlines()[0].split("-")[1].split(".")[0])
+    run_id = ckpt_pth.name
+    if len(record_type.query(run_id=run_id)) > 0:
+        previous_tested_step = int(record_type.query(run_id=run_id)[-1].step)  # type: ignore
     else:
         previous_tested_step = 0
-    print(f"Latest ckpt tested: {previous_tested_step}, Latest ckpt available: {latest_ckpt}\n")
-    if latest_ckpt >= previous_tested_step + freq:
+    logging.info(
+        f"Latest ckpt tested: {previous_tested_step}, Latest ckpt available: {latest_ckpt_step}\n"
+    )
+    if latest_ckpt_step >= previous_tested_step + freq:
         need_to_test = previous_tested_step + freq
     else:
         need_to_test = None
-
     return need_to_test
 
-def main(exp_path:str, freq:int=200000):
-    need_to_test = find_ckpt_to_test_cron(exp_path, freq)
-    if need_to_test is not None:
-        print(f"Running DPTEST for {exp_path} on ckpt-{need_to_test}...\n")
-        dptest_one_cpkt_on_all_heads(exp_path, need_to_test)
+
+def main(exp_path: str, freq: int = 200000, step: Optional[int] = None):
+    if step is None:
+        step = find_ckpt_to_test_cron(exp_path, freq, Record)
+    if step is not None:
```

Comment:
necessary?

Reply:
Yes. Since `find_ckpt_to_test_cron` can not guarantee all expected tests are finished, the `step` argument provides a backup plan to run tests on specified checkpoint. 
Or we could add a more precised checking logic in that funcion. What do you think?